server/src/main/java/org/candlepin/model/ConsumerCurator.java

```diff
@@ -280,20 +280,38 @@ public Consumer findByUuid(String uuid) {
     }
 
     @Transactional
-    public CandlepinQuery<Consumer> findByUuids(Collection<String> uuids) {
-        DetachedCriteria criteria = this.createSecureDetachedCriteria()
-            .add(Restrictions.in("uuid", uuids));
+    public Collection<Consumer> findByUuids(Collection<String> uuids) {
+        Set<Consumer> consumers = new HashSet<>();
 
-        return this.cpQueryFactory.<Consumer>buildQuery(this.currentSession(), criteria);
+        for (List<String> block : this.partition(uuids)) {
```

Comment:
Why is block unused? Isn't it then doing the same thing on every iteration?

Reply:
Fixed